### ISSUE TITLE
Can some use-case be provide for operating with "construct_base_samples_from_posterior"? 

### ISSUE THREAD

**Issue by roshni-kamath** [pytorch/botorch#461]
[Question] Can you please give some insights and paper references into how to utilize the "construct_base_samples_from_posterior" for generating initial samples?

**Comment by Balandat** [comment]
This is an internal helper function for constructing base samples for the reparameterization of samples from the posterior. You can learn more about it in https://arxiv.org/abs/1910.06403. 

What are you trying to do? Unless you want to change something about the inner workings of MC acquisition functions you should not need to use this function.

**Comment by roshni-kamath** [comment]
I want to use empirical Gaussian process distribution obtained on the 1st training process to generate random values to target training of the second  batch, and that that leads to faster convergence than sampling. Is it possible to use "construct_base_samples_from_posterior" by providing the posterior of the trained GP model?

**Comment by Balandat** [comment]
Not quite sure I fully understand. By "sampling" do you mean random sampling from the domain (e.g. as initial exploration samples)? 

If you want to do something like Thompson sampling from a GP posterior for this, you can use `MaxPosteriorSampling` in https://github.com/pytorch/botorch/blob/master/botorch/generation/sampling.py. That module also has a `BoltzmannSampling` strategy that you can use to draw samples according to a softmax-transformation of the posterior mean.

**Comment by roshni-kamath** [comment]
In the optimize_acqf function, currently, to generate batch_initial_conditions, gen_batch_initial_conditions is used. Instead of that, I want to use empirical Gaussian process distribution to generate random samples for training on the second batch. I am not sure how can I use BoltzmannSampling strategy in that case

> Not quite sure I fully understand. By "sampling" do you mean random sampling from the domain (e.g. as initial exploration samples)?
> Yes, for the initial exploration samples

In the optimize_acqf function, currently, to generate batch_initial_conditions, gen_batch_initial_conditions is used. Instead of that, I want to use empirical Gaussian process distribution to generate random samples for training on the second batch. I am not sure how can I use the BoltzmannSampling or Thompson sampling strategy in that case. FYI, I am using q-EI acquisition function

**Comment by Balandat** [comment]
Say `model` is your model from the first run, you can do the following:

```
BM = BoltzmannSampling(PosteriorMean(model), eta=0.5)
X = torch.rand(1024, d)   # Set of points on the d-dim domain to sample from
sampled_X = BM(X, n=5)  # draw n=5 samples from X according to the BoltzmannSampling weights
```

**Comment by Balandat** [comment]
Assuming this is resolved, please re-open if not.
